x/valset-pref/validator_set.go

```diff
@@ -62,10 +88,13 @@ func (k Keeper) DelegateToValidatorSet(ctx sdk.Context, delegatorAddr string, co
 			return err
 		}
 
+		// Matt: If we're truncating here, is it not true that we might be hitting the logic without
+		// fully delegating the given coin?
 		// tokenAmt takes the amount to delegate, calculated by {val_distribution_weight * tokenAmt}
 		tokenAmt := val.Weight.Mul(coin.Amount.ToDec()).TruncateInt()
```

Comment:
Great catch, yeah we need to make a running total, and delegate any surplus to one of the validators.

Reply:
how would i be storing the surplus amount here? for instance if delegateAmt = 1987.986, currently we just delegate 1987, how would i even delegate 0.986 from users Account since Delegate function takes an integer 

`Delegate(ctx sdk.Context, delAddr sdk.AccAddress, bondAmt sdk.Int, tokenSrc stakingtypes.BondStatus, validator stakingtypes.Validator, subtractAccount bool) (newShares sdk.Dec, err error)`